### PR TITLE
add user  and missing package to agent

### PR DESCRIPTION
**Pull request by bznein**
Changes needed to use the agent in enterprise

### All Submissions:

* [ ] Have you opened an Issue before filing this PR?
* [ ] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [ ] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
* [ ] Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if such).

### CODE REVIEW

**Review by chatton — APPROVED** [review]
LGTM once the branch has been merged with master 👍

File: controllers/construct/mongodbstatefulset.go
Comment (bznein) [inline controllers/construct/mongodbstatefulset.go:158]: This is sadly needed as otherwise the agent will panic with `unknown user 2000`, we will ask the agent team if they can handle it on their side in the future but for now we will have to keep it.

**Comment by bznein** [comment]
@chatton some big changes since you approved, please take another look when you have time

I already have a evg run succeeding with these same changes: https://spruce.mongodb.com/version/605b06b8850e6102a0c2d85d/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC

File: controllers/construct/mongodbstatefulset.go
Comment (chatton) [inline controllers/construct/mongodbstatefulset.go:250]: can we maybe make `/var/lib/mongodb-mms-automation/authentication/keyfile` a constant in case the filepath changes in the future?

I guess we could also do it for the `automation-mongod.conf` while we're here, what do you think?

File: controllers/construct/mongodbstatefulset.go
Comment (chatton) [reply controllers/construct/mongodbstatefulset.go:158]: should we make a ticket for this now so it doesn't get overlooked?

**Review by chatton — APPROVED** [review]
LGTM - I just had a few minor questions/comments, will leave them up to you!

File: controllers/construct/mongodbstatefulset.go
Comment (bznein) [reply controllers/construct/mongodbstatefulset.go:250]: Sure, will do

File: controllers/construct/mongodbstatefulset.go
Comment (bznein) [reply controllers/construct/mongodbstatefulset.go:158]: I think so, we need to figure out exactly what we require from the agent team to write a good ticket for them, maybe we can syncup later on zoom and write it together?

**Pull request merged**
